This patch changes the base type used in udp to that
used in later glibc versions.

--- libc/sysdeps/gnu/netinet/udp.h.orig	2007-04-10 15:46:01.000000000 +0100
+++ libc/sysdeps/gnu/netinet/udp.h	2005-05-30 12:07:58.000000000 +0100
@@ -54,21 +54,23 @@
 
 /* UDP header as specified by RFC 768, August 1980. */
 #ifdef __FAVOR_BSD
+
 struct udphdr
 {
-  uint16_t uh_sport;           /* source port */
-  uint16_t uh_dport;           /* destination port */
-  uint16_t uh_ulen;            /* udp length */
-  uint16_t uh_sum;             /* udp checksum */
+  u_int16_t uh_sport;		/* source port */
+  u_int16_t uh_dport;		/* destination port */
+  u_int16_t uh_ulen;		/* udp length */
+  u_int16_t uh_sum;		/* udp checksum */
 };
+
 #else
 
 struct udphdr
 {
-  uint16_t source;
-  uint16_t dest;
-  uint16_t len;
-  uint16_t check;
+  u_int16_t source;
+  u_int16_t dest;
+  u_int16_t len;
+  u_int16_t check;
 };
 #endif
 
